src/users/index.ts

```diff
@@ -18,36 +46,139 @@ export interface Users {
    * @returns {Promise<Record<string, Role>>} A map of role names to their respective roles.
    */
   getAssignedRoles: (userId: string) => Promise<Record<string, Role>>;
+
+  db: DBUsers;
+  oidc: OIDCUsers;
+}
+
+/** Operations supported for namespaced 'db' users.*/
+export interface DBUsers extends UsersBase {
   /**
-   * Assign roles to a user.
+   * Retrieve the roles assigned to a user.
    *
-   * @param {string | string[]} roleNames The name or names of the roles to assign.
-   * @param {string} userId The ID of the user to assign the roles to.
-   * @returns {Promise<void>} A promise that resolves when the roles are assigned.
+   * @param {string} userId The ID of the user to retrieve the assigned roles for.
+   * @returns {Promise<Record<string, Role>>} A map of role names to their respective roles.
    */
-  assignRoles: (roleNames: string | string[], userId: string) => Promise<void>;
+  getAssignedRoles: (userId: string, includePermissions?: boolean) => Promise<Record<string, Role>>;
```

Comment:
Despite this being the only optional parameter for now in this method, it could be extended at some point in the future. Can we introduce a type here, say `type GetAssignedRolesOptions = { includePermissions?: boolean }`, and have the second argument here be `opts?: GetAssignedRolesOptions`?

That way usage would be more inline with the rest of the codebase, e.g.
```typescript
.getAssignedRoles("id", { includePermissions: true })
```
and this would be easily extendable if we need to add other types of includes depending on the future complexity of the role object

Reply:
> That way usage would be more inline with the rest of the codebase

Do I gather correctly that our convention then is to group optional arguments in one optional object?
I'll update all the call sites then

Reply:
Yes, that's correct! E.g., for searching:
```typescript
res = await collection.query.near_text("A query", {
  limit: 5,
  returnMetadata: ['distance'],
})
```